docs-ref-services/servicebus.md

```diff
@@ -2,47 +2,200 @@
+message_two = Message("Second")
+
+with queue_client.get_sender() as sender:
+    sender.send(message_one)
+    sender.send(message_two)
+```
+
+### Receiving messages
+Message can be received from a queue as a continuous iterator. The default mode for message receiving is [PeekLock](https://docs.microsoft.com/rest/api/servicebus/peek-lock-message-non-destructive-read), which requires each message to be explicitly completed in order that it be removed from the queue.
+```python
+messages = queue_client.get_receiver()
+for message in messages:
+    print(message)
+    message.complete()
+```
+The service connection will remain open for the entirety of the iterator.
+If you find yourself only partially iterating the message stream, you should run the receiver in a `with` statement to ensure the connection is closed:
+```python
+with queue_client.get_receiver() as messages:
+    for message in messages:
+        print(message)
+        message.complete()
+        break
+```
+
+## Service Bus Topics and Subscriptions
+
+Service Bus topics are an abstraction on top of Service Bus Queues that
+make pub/sub scenarios easy to implement.
+
+### Create topic
+This creates a new topic within the Service Bus namespace. If a topic of the same name already exists an error will be raised. 
+```python
+sb_client.create_topic("MyTopic")
+```
+
+### Get a topic client
+A TopicClient can be used to send messages to the topic, along with other operations.
+```python
+topic_client = sb_client.get_topic("MyTopic")
+```
+
+### Create subscription
+This creates a new subscription for the specified topic within the Service Bus namespace.
+```python
+sb_client.create_subscription("MyTopic", "MySubscription")
+```
+
+### Get a subscription client
+A SubscriptionClient can be used to receive messages from the topic, along with other operations.
+```python
+topic_client = sb_client.get_subscription("MyTopic", "MySubscription")
+```
+
+## Migration from v0.21.1 to v0.50.0
+Major breaking changes were introduced in version 0.50.0.
+The original HTTP-based API is still available in v0.50.0 - however it now exists under a new namesapce: `azure.servicebus.control_client`.
+
+### Should I upgrade?
+The new package (v0.50.0) offers no improvements in HTTP-based operations over v0.21.1. The HTTP-based API is identical except that it now exists under a new namespace. For this reason if you only wish to use HTTP-based operations (`create_queue`, `delete_queue` etc) - there will be no additional benefit in upgrading at this time.
 
-Service bus namespaces created using the Azure portal after August 2014
-no longer support ACS authentication. You can create ACS compatible
-namespaces with the Azure SDK.
 
-### Shared Access Signature Authentication
+### How do I migrate my code to the new version?
+Code written against v0.21.0 can be ported to version 0.50.0 by simply changing the import namespace:
+
+```python
+from azure.servicebus.control_client import ServiceBusService
```

Comment:
Maybe add the old import statement, commented-out, above the new one to make it super clear for those who simply scan for code blocks?